core/templates/dev/head/components/question-directives/question-editor/question-editor.directive.ts

```diff
@@ -223,8 +212,23 @@ angular.module('oppia').directive('questionEditor', [
             _init();
           });
 
-          _init();
-        }
-      ]
+          ctrl.$onInit = function() {
+            if (ctrl.canEditQuestion()) {
+              EditabilityService.markEditable();
+            } else {
+              EditabilityService.markNotEditable();
+            }
+            StateEditorService.setActiveStateName('question');
+            StateEditorService.setMisconceptionsBySkill(
+              ctrl.getMisconceptionsBySkill());
+            ctrl.oppiaBlackImgUrl = UrlInterpolationService.getStaticImageUrl(
+              '/avatar/oppia_avatar_100px.svg');
+
+            ctrl.interactionIsShown = false;
+
+            ctrl.stateEditorInitialized = false;
+            _init();
```

Comment:
Hm, seems weird to call _init() in a function named $onInit(). Should we just inline _init() in here, if it's not called by anything else?

Reply:
Added comment to explain.